Prototyped Hillel's syntax suggestions in tla2tools

ahelwer opened the pull request:
cc @hwayne, freshly baked for u

Download a build of this branch here (rename .zip to .jar): 
[tla2tools.zip](https://github.com/user-attachments/files/20258205/tla2tools.zip)

This will parse the following file:
```tla
---- MODULE Test ----
ASSUME all x in {1,2,3} :
  && x != 0
  && !(x = 4)
  && || x in {1, 2}
     || x in {3, 4}
====
```
The parser changes are transparent to higher layers of the stack, so TLC also correctly checks this specification.

Changes:
1. Added `!` as synonym for `~` logical-not prefix operator (this was defined as `<BANG>` token for referring to definitions in modules imported with `INSTANCE`)
2. Added `!=` as synonym for `/=` not-equals infix operator
3. Added `in` as synonym for `\in` set-membership infix operator
4. Added `any` and `all` as synonyms for `\E` and `\A`
5. Removed existing `&&` infix operator from language and redefined it as a synonym of the `/\` logical-and infix operator (this will cause the unit tests to fail)
6. Removed existing `||` infix operator from language and redefined it as a synonym of the `\/` logical-or infix operator (this also will cause the unit tests to fail)

The only files changed by hand were `tla+.jj` and `Operators.java`; the rest are auto-generated by JavaCC.

ahelwer committed:
Fixed JavaCC warnings introduced by Unicode work
Warnings were added in PR #896
Warning types fixed:
1. Warning:  "\u2227" cannot be matched as a string literal token at
   line 1261, column 33. It will be matched as  <AND>.
2. Warning: Line 1316, Column 3: Non-ASCII characters used in regular
   expression.

There were three warnings of type (1), for <AND>, <OR>, and <IN>. The
reason for this warning was that these three tokens function both as
infix operators and also keywords. The keyword tokens were called <AND>,
<OR>, and <IN>. During the unicode work the ∧, ∨, and ∈ synonyms were
added to the <AND>, <OR>, and <IN> token definitions. However, they were
also defined in separate tokens named <op_infix_land_uc>,
<op_infix_lor_uc>, and <op_infix_in_uc> (the same as all other Unicode
operator variants). The warning was just saying that these three _uc token
definitions were being ignored in favor of <AND>, <OR>, and <IN>. The
solution was simply to delete the ignored _uc token definitions.

The warning of type (2) was because the ℕ, ℤ, and ℝ characters were
added to the general <IDENTIFIER> token regex. This was fixed by setting
the UNICODE_INPUT option to true. JavaCC now generates a "note" instead
of a warning saying to make sure to use a reader with the proper input
encoding, which indeed we do.

There is one remaining warning about a lookahead ambiguity in the CASE
statement. This warning was present before the Unicode work.

[SANY]

Signed-off-by: Andrew Helwer <ahelwer@pm.me>

ahelwer committed:
Implemented Hillel's suggested syntax variants

Signed-off-by: Andrew Helwer <ahelwer@pm.me>

hwayne commented:
Oh my!

One change that would be really annoying is `\/ -> ||`, because `||` does something in PlusCal

ahelwer commented:
Could still be done in TLA+, PlusCal parser is separate. Similar to && it would require deleting an existing operator to redefine it as a synonym of `\/`.

ahelwer commented:
Updated the parser to implement `||` as a synonym for `\/` and updated the attached build, along with the example. Also fixed a bug where `in` didn't work as an infix operator.
